bloom-housing/bloom
# feat: lottery expiration frontend and cron job

This PR addresses [#(4058)](https://app.zenhub.com/workspaces/bloom-5dc32d7144bd400001315dac/issues/gh/bloom-housing/bloom/4058)


## Description
Adds frontend logic for partner and jurisdictional admins to warn them that the lottery will expire by a certain data and to hide all date once the lottery has expired. Design/copy: https://www.figma.com/design/PAeCVANDmGdTz2Y53ouErM/Lotteries?node-id=3528-20867&t=Fn6TV5lcbTvuSqrN-0
https://www.figma.com/design/Kcsf148FqeUqMNeMo54BsW/Ad-Hoc-Feature-Enhancements?node-id=2418-16131

Creates cron job to expire lotteries. Uses an env variable to set the expiration limit in days.

Unit and E2E tests included

## How Can This Be Tested/Reviewed?
Setup: add `LOTTERY_DAYS_TILL_EXPIRY=45` to your api and partners env files

1. Sign into Partners Portal and close a listing with a lottery
2. As an admin, go to the Lottery tab of the closed listing and open the run modal. Verify that expiration statement is present at the top. Run the lottery, releasedToPartners.
3. As a partner or jurisdictional admin, go to the Lottery tab of the closed listing with a released lottery. Publish the lottery. Verify yellow warning is present at the top of the section with the expiration date included.
4. In the database, edit the closedAt date on the listing to 45 (or more) days ago. Or change your env variable to 0.
5. Sign into the Swagger UI page as an admin or jurisdictional admin. Trigger the cron job with the `listings/expireLotteries` endpoint.
6. Sign into the Partners Portal as a partner or jurisdictional admin, go to the Lottery tab of the closed listing with a ran lottery. Verify lottery is expired and no data is shown.


## Author Checklist:


## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates
